src/com/rcv/Logger.java

```diff
@@ -49,6 +50,11 @@ public static void setup(String file_output_path) throws IOException {
     // controls logging console output filters
     ConsoleHandler consoleHandler = new ConsoleHandler();
     consoleHandler.setFormatter(formatter);
+    // remove any existing handlers to prevent duplicate logging
+    for (Handler h : logger.getHandlers()) {
+      logger.removeHandler(h);
+    }
+    // add new handlers
```

Comment:
Nice.  I've run into this in python coding when running scripts from test harness so this makes sense.